src/celeritas/track/detail/TrackInitAlgorithms.cu

```diff
@@ -34,11 +35,12 @@ size_type remove_if_alive(
     StreamId stream_id)
 {
     auto start = device_pointer_cast(vacancies.data());
-    auto end = thrust::remove_if(
-        thrust::device.on(celeritas::device().stream(stream_id).get()),
-        start,
-        start + vacancies.size(),
-        IsEqual{occupied()});
+    auto end
+        = thrust::remove_if(thrust_execution_policy().on(
+                                celeritas::device().stream(stream_id).get()),
```

Comment:
This is going to be a pretty common idiom, so maybe `celeritas/ext/Thrust.device.hh` should provide a function that maps stream ID to execution space using the global device?

Reply:
I ended up adding two functions to get an execution space from a stream id, with a version check for async execution.